activemq-client/src/main/java/org/apache/activemq/ActiveMQContext.java

```diff
@@ -0,0 +1,539 @@
+        throw new UnsupportedOperationException("createDurableConsumer(topic, name, messageSelector, noLocal) is not supported");
+    }
+
+    @Override
+    public JMSConsumer createSharedDurableConsumer(Topic topic, String name) {
+        throw new UnsupportedOperationException("createSharedDurableConsumer(topic, name) is not supported");
+    }
+
+    @Override
+    public JMSConsumer createSharedDurableConsumer(Topic topic, String name, String messageSelector) {
+        throw new UnsupportedOperationException("createDurableConsumer(topic, name, messageSelector) is not supported");
+    }
+
+    @Override
+    public JMSConsumer createSharedConsumer(Topic topic, String sharedSubscriptionName) {
+        throw new UnsupportedOperationException("createSharedConsumer(topic, sharedSubscriptionName) is not supported");
+    }
+
+    @Override
+    public JMSConsumer createSharedConsumer(Topic topic, String sharedSubscriptionName, String messageSelector) {
+        throw new UnsupportedOperationException("createSharedConsumer(topic, sharedSubscriptionName, messageSelector) is not supported");
+    }
+
+    @Override
+    public QueueBrowser createBrowser(Queue queue) {
+        checkContextState();
+        try {
+            if(getAutoStart()) {
+                start();
+            }
+            return activemqSession.createBrowser(queue);
+        } catch (JMSException e) {
+            throw JMSExceptionSupport.convertToJMSRuntimeException(e);
+        }
+    }
+
+    @Override
+    public QueueBrowser createBrowser(Queue queue, String messageSelector) {
+        checkContextState();
+        try {
+            if(getAutoStart()) {
+                start();
+            }
+            return activemqSession.createBrowser(queue, messageSelector);
+        } catch (JMSException e) {
+            throw JMSExceptionSupport.convertToJMSRuntimeException(e);
+        }
+    }
+
+    @Override
+    public TemporaryQueue createTemporaryQueue() {
+        checkContextState();
+        try {
+            return activemqSession.createTemporaryQueue();
+        } catch (JMSException e) {
+            throw JMSExceptionSupport.convertToJMSRuntimeException(e);
+        }
+    }
+
+    @Override
+    public TemporaryTopic createTemporaryTopic() {
+        checkContextState();
+        try {
+            return activemqSession.createTemporaryTopic();
+        } catch (JMSException e) {
+            throw JMSExceptionSupport.convertToJMSRuntimeException(e);
+        }
+    }
+
+    @Override
+    public void unsubscribe(String name) {
+        checkContextState();
+        try {
+            activemqSession.unsubscribe(name);
+        } catch (JMSException e) {
+            throw JMSExceptionSupport.convertToJMSRuntimeException(e);
+        }
+    }
+
+    @Override
+    public void acknowledge() {
+        throw new UnsupportedOperationException("acknowledge() is not supported");
```

Comment:
Surprising that this 1.1 level functionality isnt implemented.

Reply:
Agreed that this seems simple to implement as ActiveMQSession already exposes the acknowledge method for this to leverage.

Reply:
Got it. Will do

Reply:
This is complete